### PR TITLE
 refactor(api): updated hardware information

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
## PR Checklist

- [x] Unit Tests have been added for new changes
- [x] API tests have been updated if applicable
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

## What are you changing?
Updated API to get the hardware information from AMT and added unit tests.

### CODE REVIEW

File: src/amt/connectedDevice.test.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ describe('general settings', () => {
   })
 })
 
-describe('request user conset code', () => {
+describe('user conset code', () => {
```
Comment (rsdmike) [inline src/amt/connectedDevice.test.ts:92]: conset => consent

File: src/routes/amt/getHardwareInfo.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+export async function get (guid: string): Promise<any> {
+  const response: {[key: string]: any} = {}
+  const computerSystemPackage = await devices[guid].getComputerSystemPackage()
+  if (computerSystemPackage != null) {
+    response.CIM_ComputerSystemPackage = computerSystemPackage
+  }
+  const chassis = await devices[guid].getChassis()
+  if (chassis != null) {
```
Comment (rsdmike) [inline src/routes/amt/getHardwareInfo.ts:42]: I'm not sure all these if conditions are necessary --- i think its fine to return the property set to null -- i would say thats better.

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/amt/connectedDevice.test.ts (hunk 1 above)
Comment (madhavilosetty-intel) [reply src/amt/connectedDevice.test.ts]: done

File: src/routes/amt/getHardwareInfo.ts (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/routes/amt/getHardwareInfo.ts]: done

File: src/amt/CIRAHandler.ts [hunk 3]
```diff
@@ -52,8 +53,12 @@ export class CIRAHandler {
 
   // Setup CIRA Channel
   SetupCiraChannel (socket: CIRASocket, targetPort: number): CIRAChannel {
+    console.log('cirachannel state : ', this.channel?.state)
```
Comment (rsdmike) [inline src/amt/CIRAHandler.ts:56]: use logger or remove

File: src/amt/CIRAHandler.ts [hunk 4]
```diff
@@ -63,34 +68,34 @@ export class CIRAHandler {
...
+      this.channel.closing = 1
+      APFProcessor.SendChannelClose(this.channel.socket, this.channel.amtchannelid)
+      if (this.channel.onStateChange) {
+        this.channel.onStateChange(this.channel, this.channel.state)
       }
     }
 
-    channel.sendchannelclose = (): void => {
+    this.channel.sendchannelclose = (): void => {
       console.log('Channel closed called')
```
Comment (rsdmike) [inline src/amt/CIRAHandler.ts:93]: use logger

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/amt/CIRAHandler.ts (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/amt/CIRAHandler.ts]: removed

File: src/amt/CIRAHandler.ts (hunk 4 above)
Comment (madhavilosetty-intel) [reply src/amt/CIRAHandler.ts]: removed

**Review by rsdmike — APPROVED** [review]

**Pull request merged**